addons/hr_timesheet/static/src/js/task_with_hours.js

```diff
@@ -23,14 +24,19 @@ var TaskWithHours = FieldMany2One.extend({
      * @override
      * @private
      */
-    _renderEdit: function (){
+    _renderEdit: function () {
         this.m2o_value = this._getDisplayNameWithoutHours(this.m2o_value);
         this._super.apply(this, arguments);
     },
-});
+};
+
+const TaskWithHours = FieldMany2One.extend(TaskWithHoursMixin, {});
+
+const ListTaskWithHours = ListFieldMany2One.extend(TaskWithHoursMixin, {});
 
 field_registry.add('task_with_hours', TaskWithHours);
+field_registry.add('list.task_with_hours', ListTaskWithHours);
```

Comment:
@aab-odoo We have to add separate widget for listview it is because we handled situation here(please check task: 2411029): https://github.com/odoo/odoo/blob/14.0/addons/web/static/src/js/fields/relational_fields.js#L936

What I am thinking, why not bind mousedown in many2one widget and check if there is autocomplete open(floating is true) then prevent widnow click, so if many2one is rendered inside editable list and if user write something and click outside then on first click we skip propagating window click.

But this solution will also work I am suggesting above solution so that we don't have to add separate widget for listview to handle this situation, what do you think?